source/pythonConsole.py

```diff
@@ -255,25 +255,35 @@ def updateNamespaceSnapshotVars(self):
 		Typically, used before the NVDA python console is opened, after which, calls
 		to the 'api' module will refer to this new focus.
 		"""
-		try:
-			caretPos = api.getCaretPosition()
-		except RuntimeError:
-			log.debug("Unable to set caretPos snapshot variable for python console.")
-			caretPos = None
 
-		self._namespaceSnapshotVars = {
-			"focus": api.getFocusObject(),
+		def _getCaretPos() -> textInfos.TextInfo | None:
+			try:
+				return api.getCaretPosition()
+			except RuntimeError:
+				log.debug("Unable to set caretPos snapshot variable for python console.")
+				return None
+
+		self._namespaceSnapshotVarsGetters = {
+			"focus": (lambda: api.getFocusObject()),
```

Comment:
Why not this instead of lambdas where possible?

```suggestion
			"focus": api.getFocusObject),
```